src/frontend/src/flows/addDevice/welcomeView/showVerificationCode.ts

```diff
@@ -65,79 +85,114 @@ export const showVerificationCode = async (
   tentativeRegistrationInfo: TentativeRegistrationInfo,
   credentialToBeVerified: CredentialId
 ): Promise<void> => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent(userNumber, alias, tentativeRegistrationInfo), container);
-  return init(
+  const countdown = AsyncCountdown.fromNanos(
+    tentativeRegistrationInfo.device_registration_timeout
+  );
+
+  showVerificationCodePage({
+    userNumber,
+    alias,
+    tentativeRegistrationInfo,
+    remaining: countdown.remainingFormattedAsync(),
+    cancel: () => {
+      countdown.stop();
+      window.location.reload();
```

Comment:
is this window reload new and if yes, can we spare it? 

both for performance reason and ux, I think reload should just be used in last ressort but if not new, shall be it of course

Reply:
I'm actually trying to get rid of this `reload()` 😓 But first needed to move a few things around in this PR, although I can't quite remove it just yet! But good catch